### PR TITLE
Created bypass checks for MMU

### PR DESCRIPTION
**Pull request by smwoodward**

### CODE REVIEW

File: macros/base/start_print.cfg [hunk 1]
```diff
@@ -139,19 +140,20 @@ gcode:
...
-                    MMU_PRELOAD                     # Pre-load MMU gate for initial tool if not already loaded (default T0).
-                {% endif %}
-            {% else %}
-                {% if printer.mmu.tool|int != INITIAL_TOOL %}
-                    MMU_EJECT                       # Eject bad tool
-                    MMU_SELECT TOOL={INITIAL_TOOL}  # Select MMU gate for initial tool (default T0).
-                    MMU_PRELOAD                     # Pre-load MMU gate for initial tool(default T0).
+            {% if INITIAL_TOOL != -2 %}
+                {% if printer.mmu.filament != "Loaded" %}
+                    {% if printer.mmu.tool|int != INITIAL_TOOL and INITIAL_TOOL != -2 %}
```
Comment (Benoitone) [inline macros/base/start_print.cfg:145]:
```suggestion
                    {% if printer.mmu.tool|int != INITIAL_TOOL %}
```

File: macros/base/start_print.cfg [hunk 2]
```diff
@@ -139,19 +140,20 @@ gcode:
...
+                        MMU_SELECT TOOL={INITIAL_TOOL}
+                        MMU_PRELOAD                     # Pre-load MMU gate for initial tool if not already loaded (default T0).
+                    {% endif %}
+                {% else %}
+                    {% if printer.mmu.tool|int != INITIAL_TOOL  and INITIAL_TOOL != -2 %}
```
Comment (Benoitone) [inline macros/base/start_print.cfg:150]:
```suggestion
                    {% if printer.mmu.tool|int != INITIAL_TOOL %}
```

File: macros/base/start_print.cfg [hunk 3]
```diff
@@ -139,19 +140,20 @@ gcode:
...
+                        MMU_EJECT                       # Eject bad tool
+                        MMU_SELECT TOOL={INITIAL_TOOL}  # Select MMU gate for initial tool (default T0).
+                        MMU_PRELOAD                     # Pre-load MMU gate for initial tool(default T0).
+                    {% endif %}
                 {% endif %}
-            {% endif %}
+            {% endif %}
```
Comment (Benoitone) [inline macros/base/start_print.cfg:156]:
```suggestion
            {% endif %}
```

File: macros/base/start_print.cfg [hunk 4]
```diff
@@ -352,8 +354,10 @@ gcode:
     # Load intial tool in the MMU if needed
     {% if klippain_mmu_enabled %}
         {% if printer.mmu.enabled %}
-            MMU_CHANGE_TOOL TOOL={INITIAL_TOOL} STANDALONE=1
-            _MMU_ERROR_CHECK2    # Second error check. If an error occur when loading the first tool. To stop the START_PRINT macro process
+            {% if INITIAL_TOOL != -2 %}
```
Comment (Benoitone) [inline macros/base/start_print.cfg:357]:
```suggestion
           {% if INITIAL_TOOL != -2 or printer.mmu.tool|int != -2 %}
```

**Review by Benoitone — CHANGES_REQUESTED** [review]
Thanks and can you check this changes?

**Comment by Benoitone** [comment]
But maybe with only
{% if printer.mmu.enabled and printer.mmu.tool|int != -2 %}
in the differents macros is enough

**Comment by smwoodward** [comment]
That looks good to me.  I'm not a developer by trade so I know there are always better ways to handle things.

**Comment by Benoitone** [comment]
> That looks good to me.  I'm not a developer by trade so I know there are always better ways to handle things.

I'm not developer too 😀
Based on your work I push a new branch for testing a simplified things.
Can you have a look on it?
https://github.com/Frix-x/klippain/tree/Happy_Hare_Benoit

**Pull request closed without merging**